File: client/src/models/Invitation.ts
<<<<<<< HEAD
<<<<<<< HEAD
import { InvitationStatus } from "./enums/InvitationStatus";
=======
import type { InvitationStatus } from "./InvitationStatus";
>>>>>>> 935d6e8 (Fix various code smells)
=======
import type { InvitationStatus } from "./enums/InvitationStatus";
>>>>>>> 3cfff02 (linting)

export interface Invitation {
  id: number;
  createdDateTime: string;
  modifiedDateTime: string;
  status: InvitationStatus;
  meetingId: number;
  inviterId: number;
  inviteeId: number;
}

File: server/app/src/main/java/lt/gathertime/server/entity/Invitation.java
package lt.gathertime.server.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lt.gathertime.server.enums.InvitationStatus;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Invitation {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    protected Long id;

    @Column(name = "created_date_time")
    protected LocalDateTime createdDateTime;

    @Column(name = "modified_date_time")
    protected LocalDateTime modifiedDateTime;

    @Enumerated(EnumType.STRING)
    private InvitationStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "meeting_id")
    private Meeting meeting;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "inviter_id")
    private User inviter;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "invitee_id")
    private User invitee;
}
